### ISSUE TITLE
d&b amplifiers

### ISSUE THREAD

**Issue by pdalgety** [bitfocus/companion-module-requests#538]
a series of buttons for scene recalls and mutes for d&b D series and other amplifiers

**Comment by daniep01** [comment]
Is a specification for the protocol available?

**Comment by pdalgety** [comment]
Hi Peter

I am contacting d&b via Autograph

Regards

Patrick


Patrick Dalgety
Production & Technical Management - Event & Venue Logistics
Rigging and Stage Design - Sound, Lighting & AV design
Mob - +44 7711 095795
Email - ***@***.******@***.***>

**Comment by bangadrum** [comment]
From  https://groups.google.com/g/qlab/c/9JmXaH5RduI

Applescript:
do shell script "echo 'COMMAND_STRING' | xxd -r -p | nc xxx.xxx.xxx.xxx 30013"

Available command strings:
MUTE 
Mute A:   3b00010000001b01000100000012000006c410008205000400020101
UnMute A: 3b00010000001b01000100000012000006c810008205000400020102
Mute B:   3b00010000001b01000100000012000006c510010205000400020101
UnMute B: 3b00010000001b01000100000012000006c910010205000400020102
Mute C:   3b00010000001b01000100000012000006c610018205000400020101
UnMute C: 3b00010000001b01000100000012000006ca10018205000400020102
Mute D:   3b00010000001b01000100000012000006c710020205000400020101
UnMute D: 3b00010000001b01000100000012000006cb10020205000400020102
POWER 
Power On:   3b00010000001c01000100000013000006fb1000010000040002010001   
Power Off:  3b00010000001c01000100000013000006fb1000010000040002010000 
Pre-formatted OCA strings for the "Load" command of the Presets Agent. 
The preset number is contained in the last byte!
Load Preset 1: 3b00010000001b010001000000120000013000002713000300030101
Load Preset 2: 3b00010000001b010001000000120000013000002713000300030102

**Comment by roomac** [comment]
Here is the JS library for the OCA (aka AES70) protocol: https://github.com/DeutscheSoft/AES70.js

**Comment by roomac** [comment]
The shell commands above work OK for me as I just need to mute/unmute an individual channel. I had a quick check of the AES70 library using their test code and it connects to our 10D amp and spits out all the objects (very lengthy!).

**Comment by pdalgety** [comment]
Does that code just get copied into a button?

**Comment by roomac** [comment]
Yes, just the bit within the double quotes above and substitute the appropriate command, e.g. `echo '3b00010000001b01000100000012000006c610018205000400020101' | xxd -r -p | nc 192.168.0.19 30013` to mute channel C

**Comment by echalupka** [comment]
Adding a +1 on this request. I believe all of the documentation is supplied at this point, just need to move to a module. In the meantime of anyone getting started on doing this with internal scripts; I highly advise taking advantage of internal variables for your amp IP addresses, mute, and power commands. Severely sped up the process on this side.

**Comment by AlexK-99** [comment]
Here is a link to the module I made for the d&B 5D amp. I want see if i can make other amplifiers work but i dont have access to other amplifiers at the moment 
[https://github.com/AlexK-99/companion-module-d-und-b-5d](https://github.com/AlexK-99/companion-module-d-und-b-5d)

**Comment by echalupka** [comment]
Alex,
I have 30Ds and 40Ds. Let me know if there's a way I can help test things.
Eric

**Comment by AlexK-99** [comment]
@echalupka Yes, you can.

Please download the following github repo and run it with node js.  [https://github.com/AlexK-99/check-aes70.git](https://github.com/AlexK-99/check-aes70.git
)
But before you run that please do (npm install) and install the dependencies.

In the file checkaes70.js there are 2 variables once for HOST(IP) and once for PORT(50014 for D40/40D/5D and 30013 for all others) please change them according to your amps

After you have executed this, a file with the name msgdata.json should be created. Please could you make this available for me to download.

**Comment by echalupka** [comment]
@AlexK-99 apologies for being a newbie here -- Could we set up a zoom call or something else to walk through these steps? I've never run anything with node js or performed npm install to install dependencies.

Eric

**Comment by AlexK-99** [comment]
> @AlexK-99 apologies for being a newbie here -- Could we set up a zoom call or something else to walk through these steps? I've never run anything with node js or performed npm install to install dependencies.
> 
> Eric

are you on windows or mac then i can make suitable executable file

**Comment by echalupka** [comment]
Mac
________________________________

**Comment by AlexK-99** [comment]
@echalupka Is your Mac already ARM or still Intel?

**Comment by echalupka** [comment]
Intel
________________________________

**Comment by AlexK-99** [comment]
@echalupka hi Eric,

[https://workupload.com/file/n84y2ZhQXK4](https://workupload.com/file/n84y2ZhQXK4)

Here is the link to the application, simply unzip and then run checkaes70.bin and then provide me with the msgdata.json for download. You can change the HOST (ip) and PORT (50014 for D40/40D/5D and 30013 for all others) in the config.json depending on the amp type.

**Comment by echalupka** [comment]
When I double-click "checkaes70" (not checkaes70.js), a text document opens up.  Is there a specific application I should be "running" this with?

Apologies,
Eric

**Comment by AlexK-99** [comment]
> When I double-click "checkaes70" (not checkaes70.js), a text document opens up. Is there a specific application I should be "running" this with?
> 
> Apologies, Eric

Actually, by default the .bin should open in the terminal. Try to start it via terminal. On the Mac where I tested it I just had to double click to start it

**Comment by tm1000** [comment]
@AlexK-99 i have 3x 10d and a 30d. I'm also a software engineer by day. I'll give this a whirl and let you know. Would be great to get into core companion

**Comment by AlexK-99** [comment]
@tm1000 i am very curious about your results.

**Comment by tm1000** [comment]
@AlexK-99 Hi there.

I got it connected (its a little finicky but just needs some work)

The mute toggle, on, off work on all the 10d and 30ds I have

Unfortunately the power on/off/toggle does not work at all.

I'll start bringing this over to your repo moving forward. We can work on this together
